server/auth/providers/google.js

```diff
@@ -86,12 +86,12 @@ if (GOOGLE_CLIENT_ID) {
     )
   );
 
-  router.get(
-    "google",
+  router.get("google", (ctx) =>
     passport.authenticate(providerName, {
       accessType: "offline",
       prompt: "select_account consent",
-    })
+      hd: ctx.request.query?.hd,
```

Comment:
I don't have a distinct reason why it's a problem, but it feels risky to forward this through – could be an avenue for modifying query values such as `state` potentially?

It's probably safer to pass an `id` to `/auth/google` and then re-query the db to get the hosted domain.